src/particle.cpp

```diff
@@ -198,11 +205,20 @@ void Particle::event_advance()
   double distance = std::min(boundary().distance, collision_distance());
 
   // Advance particle in space and time
-  for (int j = 0; j < n_coord(); ++j) {
-    coord(j).r += distance * coord(j).u;
-  }
+  this->move_distance(distance);
```

Comment:
After spending some time with this, I figured out that this change (event though it doesn't change the logic) was the cause of the failure in the `surface_source` test. Can you revert this change for now (along with the associated changes you made for that test)?

Reply:
Okay. We use move_distance also in the time_cutoff to move the particle back to the place where it was killed. For now, I have only changed it at the point you specified. To be honest, I still don't understand exactly what this changes :sweat_smile:. 
I added a comment that we can set it back to this->move_distance(distance) in the future after fixing the surface source.